observability/WFLY-18582_add-prometheus-endpoint-to-micrometer-extension.adoc

```diff
@@ -0,0 +1,121 @@
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.redhat.com/browse/WFLY-18582[WFLY-18582]
+
+=== Stability Level
+// Choose the planned stability level for the proposed functionality
+* [ ] Experimental
+
+* [X] Preview
+
+* [ ] Community
+
+* [ ] default
+
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+=== Testing By
+* [x] Engineering
+
+* [ ] QE
+
+=== Affected Projects or Components
+
+Only the Micrometer subsystem and related tests will be affected
+
+=== Other Interested Projects
+
+- OpenShift has an interest in Prometheus metrics
+
+=== Relevant Installation Types
+* [x] Traditional standalone server (unzipped or provisioned by Galleon)
+
+* [ ] Managed domain
+
+* [x] OpenShift s2i
+
+* [x] Bootable jar
+
+== Requirements
+
+=== Hard Requirements
+
+* Existing servers, if configured to use Micrometer (e.g., the `endpoint` property has been set), will be migrated to be
+an OTLP-based configuration. Since OTLP is the only current option, there is no ambiguity in this migration.
+* Existing servers with the extension enabled but with no endpoint set will not be changed. That is to say, Micrometer
+will continue to use a no-op registry.
+* Support for a new registry, `PrometheusRegistry` will be added.
+** It can be added as the only registry, or it can be added _in addition to_ the OTLP Registry
+** The registry will expose two model attribute:
+*** context: the context (e.g., `/prometheus`) on which the metrics will be exposed
+*** security-enabled: whether or not to secure the context using the user and roles defined in the application server
+** If the user would like to use `/metrics`, the WildFly metrics extension must be disabled first to avoid a naming conflict.
+
+=== Non-Requirements
+
+* Adding other arbitrary `MeterRegistry` types
+
+== Backwards Compatibility
+
+* Existing servers should continue to operate as is, using any existing OTLP configuration without manual intervention.
+
+=== Default Configuration
+
+* The default context for the Prometheus scrape will be undefined and must be specified by the user when enabling support (`context`)
+* The context will be secured (`security-enabled`) by default
+
+=== Importing Existing Configuration
+
+* See above.
+
+=== Deployments
+
+=== Interoperability
+
+== Security Considerations
+
+If `security-enabled` is set to true:
+* The endpoint will use basic authorization to control access to the endpoint itself
+* As each metric is read, the system will verify that the authenticated user has access to that particular metric before returning the value.
+** All metrics that are disallowed for the user per the security policy will return a zero
+
+== Test Plan
+
+* Tests will be added to `testsuite/integration/microprofile/`
+** Verify basic Prometheus support
+** Verify security can be enabled
+
+== Community Documentation
+
+The WildFly Administration Guide will be updated to include this new support.
```

Comment:
Do we want to add more detail here or simply just address this in the implementation PR with documentation>

Reply:
As I understand the process, it is sufficient here to note that the admin guide will be updated, but that the exact nature of the change need not be spelled out in detail. I'm happy to update this, though, if my understanding is incorrect.